backend/module/chaos/serverexperimentation/xds/server_runtime_generation.go

```diff
@@ -0,0 +1,237 @@
+
+		switch httpFaultConfig.GetFaultTargeting().GetEnforcer().(type) {
+		case *serverexperimentation.FaultTargeting_DownstreamEnforcing:
+			// Abort Egress Fault
+			percentageKey = fmt.Sprintf(HTTPPercentageForEgress, egressPrefix, upstreamCluster)
+			faultKey = fmt.Sprintf(HTTPStatusForEgress, egressPrefix, upstreamCluster)
+
+		case *serverexperimentation.FaultTargeting_UpstreamEnforcing:
+			// Abort Internal Fault for all downstream services
+			if downstreamCluster == "" {
+				percentageKey = fmt.Sprintf(HTTPPercentageWithoutDownstream, ingressPrefix)
+				faultKey = fmt.Sprintf(HTTPStatusWithoutDownstream, ingressPrefix)
+			} else {
+				// Abort Internal Fault for a given downstream services
+				percentageKey = fmt.Sprintf(HTTPPercentageWithDownstream, ingressPrefix, downstreamCluster)
+				faultKey = fmt.Sprintf(HTTPStatusWithDownstream, ingressPrefix, downstreamCluster)
+			}
+
+		default:
+			return "", 0, "", 0, fmt.Errorf("unknown enforcer %v", httpFaultConfig)
+		}
+
+	case *serverexperimentation.HTTPFaultConfig_LatencyFault:
+		latency := httpFaultConfig.GetLatencyFault()
+		percentageValue = latency.GetPercentage().GetPercentage()
+		faultValue = latency.GetLatencyDuration().GetFixedDurationMs()
+
+		switch httpFaultConfig.GetFaultTargeting().GetEnforcer().(type) {
+		case *serverexperimentation.FaultTargeting_DownstreamEnforcing:
+			// Latency Egress Fault
+			percentageKey = fmt.Sprintf(LatencyPercentageForEgress, egressPrefix, upstreamCluster)
+			faultKey = fmt.Sprintf(LatencyDurationForEgress, egressPrefix, upstreamCluster)
+
+		case *serverexperimentation.FaultTargeting_UpstreamEnforcing:
+			// Latency Internal Fault for all downstream services
+			if downstreamCluster == "" {
+				percentageKey = fmt.Sprintf(LatencyPercentageWithoutDownstream, ingressPrefix)
+				faultKey = fmt.Sprintf(LatencyDurationWithoutDownstream, ingressPrefix)
+			} else {
+				// Latency Internal Fault for a given downstream services
+				percentageKey = fmt.Sprintf(LatencyPercentageWithDownstream, ingressPrefix, downstreamCluster)
+				faultKey = fmt.Sprintf(LatencyDurationWithDownstream, ingressPrefix, downstreamCluster)
+			}
+
+		default:
+			return "", 0, "", 0, fmt.Errorf("unknown enforcer %v", httpFaultConfig)
+		}
+
+	default:
+		return "", 0, "", 0, fmt.Errorf("unknown fault type %v", httpFaultConfig)
+	}
+
+	return percentageKey, percentageValue, faultKey, faultValue, nil
+}
+
+func maybeUnmarshalFaultTest(experiment *experimentation.Experiment, httpFaultConfig *serverexperimentation.HTTPFaultConfig) bool {
+	err := experiment.GetConfig().UnmarshalTo(httpFaultConfig)
+	if err != nil {
+		return false
+	}
+
+	switch httpFaultConfig.GetFault().(type) {
+	case *serverexperimentation.HTTPFaultConfig_AbortFault:
+		return true
+	case *serverexperimentation.HTTPFaultConfig_LatencyFault:
+		return true
+	default:
+		return false
+	}
+}
+
+func getClusterPair(httpFaultConfig *serverexperimentation.HTTPFaultConfig) (string, string, error) {
```

Comment:
This is also used by ECDS. Maybe keep it in xds_cache?

Reply:
I'm a little torn here since I want to keep xds_cache fault config agnostic.